gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/avro/HiveAvroCopyEntityHelper.java

```diff
@@ -72,6 +74,40 @@ public static void updateTableAttributesIfAvro(Table targetTable, HiveCopyEntity
   }
 
   /**
+   * Currently updated the {@link #HIVE_TABLE_AVRO_SCHEMA_URL} location for new hive partitions
+   * @param targetTable, new Table to be registered in hive
+   * @param sourcePartitions, source partitions
+   * @throws IOException
+   */
+  public static void updatePartitionAttributesIfAvro(Table targetTable, Map<List<String>, Partition> sourcePartitions, HiveCopyEntityHelper hiveHelper) throws IOException {
+    if (!isHiveTableAvroType(targetTable)) {
+      return;
+    }
+
+    // need to update the {@link #HIVE_TABLE_AVRO_SCHEMA_URL} location
+    for (Map.Entry<List<String>, Partition> partition : sourcePartitions.entrySet()) {
+      String oldAvroSchemaURL = partition.getValue().getTPartition().getSd().getSerdeInfo().getParameters().get(HIVE_TABLE_AVRO_SCHEMA_URL);
+      if (oldAvroSchemaURL != null) {
+
+        Path oldAvroSchemaPath = new Path(oldAvroSchemaURL);
+        URI sourceFileSystemURI = hiveHelper.getDataset().getFs().getUri();
+
+        if (PathUtils.isAbsoluteAndSchemeAuthorityNull(oldAvroSchemaPath) || (oldAvroSchemaPath.toUri().getScheme().equals(sourceFileSystemURI.getScheme())
+            && oldAvroSchemaPath.toUri().getAuthority().equals(sourceFileSystemURI.getAuthority()))) {
+
+          String newAvroSchemaURL = hiveHelper.getTargetPathHelper()
+              .getTargetPath(oldAvroSchemaPath, hiveHelper.getTargetFileSystem(), Optional.<Partition>absent(), true)
+              .toString();
+
```

Comment:
instead of relying on the table's newAvroSchemaURL, I am now calculating newAvroSchemaURL for every partition. The reason being is every partition has its own schema file, though they all might be same.